Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Repository.rst

```diff
@@ -170,3 +170,18 @@ Example to use the fallback to the default language when working with overlays:
 ..  literalinclude:: _Repository/_LanguageAspect.php
     :language: php
     :caption: EXT:my_extension/Classes/Repository/MyRepository.php
+
+..  _extbase-repository-debug-query:
+
+Debugging an Extbase query
+==========================
+
+When using complex queries in Extbase repositories it sometimes comes handy
+to debug them using the Extbase debug utilities.
```

Comment:
It's a good thing we can take this into our documentation, the Stackoverflow posts to this are a bit confusing and misleading. Thanks!

Did you by chance ask Stefan Bürk whether there's another preferred way to debug/return a query, avoiding an internal method? Using xdebug AFAIK doesn't yield the SQL structure from variable inspection.

Reply:
As this is only used during debugging there is not harm. Its just that you should not depend on it in production context